package-drone/src/main/java/de/dentrassi/pm/jenkins/DroneRecorder.java

```diff
@@ -445,33 +402,23 @@ public boolean isEmptyArchive ()
         }
 
         @Override
-        public Void invoke ( final File basedir, final VirtualChannel channel ) throws IOException, InterruptedException
+        public UploaderResult invoke ( final File basedir, final VirtualChannel channel )
         {
+            UploaderResult uploadResult = new UploaderResult ();
             final FileSet fileSet = Util.createFileSet ( basedir, this.includes, this.excludes );
             fileSet.setDefaultexcludes ( this.defaultExcludes );
 
             final String[] includedFiles = fileSet.getDirectoryScanner ().getIncludedFiles ();
             if ( includedFiles.length == 0 )
             {
-                this.emptyArchive = true;
-                // nothing to upload
+                uploadResult.setEmptyUpload ( true );
+                this.listener.error(Messages.DroneRecorder_noMatchFound (this.includes)); // nothing to upload
```

Comment:
If allowEmptyArchive is false it should be an error because it's the reason the build failed. In the other case it should be a warning.
For failsAsUpload it would actually be the same.